Bump to 21w06a

gabizou opened the pull request:
Just to show the changes from 1.16.5 to Minecraft Snapshot 21w06a.

Zidane committed:
Initial changes for 21w03a.

Signed-off-by: Chris Sanders <zidane@spongepowered.org>

Zidane committed:
More compile error fixes.

Signed-off-by: Chris Sanders <zidane@spongepowered.org>

Zidane committed:
Likely not right on the effects, Gabe will need to know for sure.

Signed-off-by: Chris Sanders <zidane@spongepowered.org>

Zidane committed:
More error fixes.

Signed-off-by: Chris Sanders <zidane@spongepowered.org>

gabizou committed:
Update setBlockState logic with relation to tile entity removals.

gabizou committed:
Snapshot: Update setBlockState logic to match new snapshot code

Signed-off-by: Gabriel Harris-Rouquette <gabizou@me.com>

gabizou committed:
Snapshot: Compiles and shadows are retargeted for 21w05b.

Signed-off-by: Gabriel Harris-Rouquette <gabizou@me.com>

gabizou committed:
Merge api-8 in for updates towards snapshot updates

Signed-off-by: Gabriel Harris-Rouquette <gabizou@me.com>

Zidane commented on src/accessors/java/org/spongepowered/common/accessor/server/level/ServerPlayerGameModeAccessor.java:
```diff
@@ -36,4 +37,6 @@
 
     @Accessor("destroyPos") BlockPos accessor$destroyPos();
 
+    @Accessor("level") ServerLevel accessor$getLevel();
```
Violates accessor naming policy, should be accessor$level()

Zidane commented on src/accessors/java/org/spongepowered/common/accessor/world/level/LevelAccessor.java:
```diff
@@ -24,18 +24,11 @@
+import org.spongepowered.asm.mixin.Mixin;
 
 @Mixin(Level.class)
 public interface LevelAccessor {
```
This accessor needs to be deleted if nothing is left in it.

Zidane commented on src/accessors/java/org/spongepowered/common/accessor/world/level/block/entity/AbstractFurnaceBlockEntityAccessor.java:
```diff
@@ -51,4 +57,9 @@
 
+    @Invoker("canBurn")
+    static boolean invoker$canBurn(@Nullable Recipe<?> var0, NonNullList<ItemStack> var1, int var2) {
+        throw new UnsupportedOperationException("Untransformed shadow");
```
There is an exception to throw for this, UntransformedInvokerError

gabizou replied:
Blame copy paste from the shadow that existed before

Zidane commented on src/accessors/java/org/spongepowered/common/accessor/world/level/entity/EntityTickListAccessor.java:
```diff
@@ -0,0 +1,37 @@
+@Mixin(EntityTickList.class)
+public interface EntityTickListAccessor {
+
+    @Accessor("active") Int2ObjectMap<Entity> accessor$getActive();
```
accessor$active()

Zidane commented on src/accessors/java/org/spongepowered/common/accessor/world/level/entity/PersistentEntitySectionManagerAccessor.java:
```diff
@@ -0,0 +1,38 @@
+@Mixin(PersistentEntitySectionManager.class)
+public interface PersistentEntitySectionManagerAccessor<T extends EntityAccess> {
+
+    @Accessor("sectionStorage") EntitySectionStorage<T> accessor$getSectionStorage();
```
accessor$sectionStorage()

Zidane commented on src/accessors/java/org/spongepowered/common/accessor/world/level/entity/TransientEntitySectionManagerAccessor.java:
```diff
@@ -0,0 +1,37 @@
+@Mixin(TransientEntitySectionManager.class)
+public interface TransientEntitySectionManagerAccessor<T extends EntityAccess>{
+
+    @Accessor("sectionStorage") EntitySectionStorage<T> accessor$getSectionStorage();
```
accessor$sectionStorage()

kashike commented on src/accessors/java/org/spongepowered/common/accessor/world/level/levelgen/NoiseGeneratorSettingsAccessor.java:
```diff
@@ -35,8 +35,15 @@
 
-    @Invoker("<init>") static NoiseGeneratorSettings invoker$new(final StructureSettings structureSettings, final NoiseSettings noiseSettings,
-            final BlockState defaultBlock, final BlockState defaultFluid, final int bedrockRoofY, final int bedrockFloorY, final int seaLevel, final boolean disableMobGeneration) {
+    @Invoker("<init>") static NoiseGeneratorSettings invoker$new(
```
Missing `final` for parameters now.

gabizou committed:
Update implementation to run 21w06a
Some aspects of the implementation need reviewing from people like @Zidane,
but otherwise the client will run in IDE and mixins apply.

Signed-off-by: Gabriel Harris-Rouquette <gabizou@me.com>
